apps/designer/app/designer/shared/assets/sanitize-s3-key.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * TEMPORARY: This function is a temporary fix for https://github.com/remix-run/remix/issues/4443
+ *
+ * Based on this: https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html
+ * Characters that might require special handling
+ * ASCII character ranges 00–1F hex (0–31 decimal) and 7F (127 decimal)
+ * "&","$", "@", "=", ";" "/", ":", "+", Significant sequences of spaces might be lost in some uses (especially multiple spaces)
+ * ",", "?"
+ * Characters to avoid
+ * Non-printable ASCII characters (128–255 decimal characters)
+ * "\", "{", "^", "}", "%", "`", "]", ', ", “, ”, ">", "[", "~",  "<", "#", "|"
+ **/
+
+// eslint-disable-next-line no-control-regex
+const specialRegexps = [/[\x00-\x1F\x7F]+/g, /[&$@=;/:+\s,?]+/g];
+
+// eslint-disable-next-line no-control-regex
+const avoidRegexps = [/[\x80-\xFF]+/g, /[\\{^}%`\]'"“”>[~<#|]+/g];
+
+const allRegexps = [...specialRegexps, ...avoidRegexps];
+
+const REPLACE_CHAR = "_";
+
+export const sanitizeS3Key = (str: string): string => {
```

Comment:
should this function be in asset-uploader package?

Reply:
oh you have it twice in the PR now???

Reply:
I wrote in the commit description and in duplicate comments that it's temporary, and since remix will fix it we will just remove it from UI side. No need to share tmp things

Reply:
what's the problem with sharing it until it's not fixed? it can be imported from the package just like any other function. its a non-trivial function with lots of logic hidden in the regexes, it needs to have tests etc, what stops you from reusing it? Just the fact it will be removed from the designer means you can remove the usage of it from the  designer.

Reply:
What package I should put it in? I always prefer duplication instead of referencing other packages.
I really don't like the idea of putting the same piece of code into a shareable place. Here is obviously temporary function wich is not needed to be in public interface of any package.